### DISCUSSION TITLE
Using `@layer` with `@tailwind` directives

### DISCUSSION THREAD

**Discussion by jordanthornquest** [tailwindlabs/tailwindcss#15006] [Help] [answered]
Hello folks,

I'm working with a pretty standard Vite + Tailwind setup. I'm attempting to set up a library called [PrimeReact](https://primereact.org/tailwind/#csslayer), which recommends configuring the `@tailwind` directives within a couple `@layer` directives.

My current, working Tailwind configuration within my `index.css` looks like this:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

However, when attempting to convert my Tailwind configuration to this:

```css
@layer tailwind-base, primereact, tailwind-utilities;

@layer tailwind-base {
  @tailwind base;
}

@layer tailwind-utilities {
  @tailwind components;
  @tailwind utilities;
}
```

My existing styles in my application break. Any idea why that might be?

**Comment by wongjn** [comment]
Please consider providing an example project that demonstrates the breaking of styles.

**Comment by jordanthornquest** [comment, marked as answer]
Following up on this in case anyone has a similar error.

The problem is because I was loading Tailwind incorrectly. I was encountering an error similar to [this discussion](https://github.com/tailwindlabs/tailwindcss/discussions/7343#discussioncomment-2159920), where a user was "running Tailwind twice". I was loading `@tailwind` directives in two different CSS files, which were each being loaded by separate `.tsx` component files. 

The solution was to follow the [Using with Preprocessors](https://tailwindcss.com/docs/using-with-preprocessors) best practices and to `@import` all of my smaller CSS files into a single CSS file, which I load in the top-level `main.tsx` component of my application. I also made sure to only use `@tailwind` directives once, importing them before the rest of my CSS. Once I did this, I was able to use the `@layer` syntax above without issue.
